jina/drivers/cache.py

```diff
@@ -55,11 +50,21 @@ def on_miss(self, req_doc: 'Document', value: str) -> None:
     def on_hit(self, req_doc: 'Document', hit_result: Any) -> None:
         """Call when cache is hit for a document.
 
-        :param req_doc: the document in the request and hitted in the cache
+        :param req_doc: the document in the request and hit in the cache
         :param hit_result: the hit result returned by the cache
         """
         pass
 
+    @staticmethod
+    def hash_doc(doc: 'Document', fields: List[str]) -> str:
+        """Calculate hash by which we cache.
+
+        :param doc: the Document
+        :param fields: the list of fields
+        :return: the hash value of the fields
+        """
+        return str(hash(tuple((doc.get_attrs(*fields).values()))))
```

Comment:
depending on the context, `hash` here may not be the good function, it may give different values on different runs.  as @JoanFM pointed out before

Reply:
Pointed out where? His comment above is just about hashing a tuple. Should we use one of these https://docs.python.org/3.7/library/hashlib.html ? Maybe sha256?

Reply:
Adapted to use that. Recheck